src/app/util/mock/MockNodeConfigImpl.h

```diff
@@ -0,0 +1,25 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/mock/MockNodeConfig.h>
+
+const chip::Test::MockNodeConfig & GetMockNodeConfig();
+
```

Comment:
Reset should go in here as well if we have one, but since we use a pointer now, we should probably just state we allow nullptr and remove the reset implementation.

Since we pass a pointer, lifetime is not as clear. I would suggest a RAII object that sets a config in a constructor and resets it back in a destructor, so that we can explicitly set the lifetime of config.